topic-operator/src/main/java/io/strimzi/operator/topic/BatchingTopicController.java

```diff
@@ -1169,6 +1171,15 @@ private void skipNonAlterableConfigs(Set<AlterConfigOp> alterConfigOps) {
                 alterConfigOps.removeIf(op -> !alterablePropertySet.contains(op.configEntry().name()));
             }
         }
+
+        var unalterableConfigs = config.unalterableTopicConfig();
+        if (unalterableConfigs != null && alterConfigOps != null && !unalterableConfigs.isEmpty()) {
+            if (!unalterableConfigs.equalsIgnoreCase("NONE")) {
```

Comment:
I consciously excluded the ALL case here, because I didn't see why we would want to block all configs, but could be added of course